### PR TITLE
Feat: improve SSL error logging/unwrapping

### PR DESCRIPTION
**Pull request by kares**
To be able to easily log full Java SSL exception details, we first move the SSL builder to Java.

otherwise this is ~ same as https://github.com/logstash-plugins/logstash-input-beats/pull/405 : 
> The unwrapping at the Java level is for [exceptions wrapped by Netty](https://github.com/netty/netty/blob/netty-4.1.49.Final/handler/src/main/java/io/netty/handler/ssl/JdkSslServerContext.java#L288).

> Full exception details will be logged at debug level from the Java side - since we seem to prefer (manual) exception logging at the plugin level. We also make sure to log cause, if any, on the Ruby side which now catches all (expected) Java exceptions.

**NOTE:** we also avoid installing the bouncy-castle security provider at runtime, [done previously](https://github.com/logstash-plugins/logstash-input-tcp/blob/v6.0.10/lib/logstash/inputs/tcp/compat_ssl_options.rb#L87)

resolves https://github.com/logstash-plugins/logstash-input-tcp/issues/177

### CODE REVIEW

File: src/main/java/org/logstash/tcp/SslContextBuilder.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        final ArrayList<Certificate> certificates = new ArrayList<>();
+        final FileInputStream fis = new FileInputStream(file);
+        try {
+            while (fis.available() > 0) {
+                try {
+                    certificates.add( factory.generateCertificate(fis) );
+                } catch (CertificateException ex) {
+                    Throwable cause = ex.getCause();
+                    if (cause != null && "Empty input".equals(cause.getMessage())) {
+                        continue;
```
Comment (robbavey) [inline src/main/java/org/logstash/tcp/SslContextBuilder.java:175]: Is there a case for `debug` or `trace` logging upon receiving this error, or is it always benign?

**Review by robbavey — APPROVED** [review]
minor debug logging question, otherwise LGTM

File: src/main/java/org/logstash/tcp/SslContextBuilder.java (hunk 1 above)
Comment (kares) [reply src/main/java/org/logstash/tcp/SslContextBuilder.java:175]: good call, I simply converted from the .rb parts and did not revisit with potential logging.
the whole logic seems a bit unusual - would rather just use `generateCertificates` to read multiple entries.
anyhow, let's keep that as before and I am going to add a debug line around the caught "empty" exception.

Comment (kares) [reply src/main/java/org/logstash/tcp/SslContextBuilder.java:181]: added debug logging + an extra check when expecting to read at least one certificate from `ssl_cert`
also worth noting (in the changelog) we no longer have the side effect of setting up the BC provider globally.

**Review by robbavey — APPROVED** [review]
LGTM

**Pull request merged**